icu4c/source/i18n/plurrule.cpp

```diff
@@ -796,53 +843,79 @@ AndConstraint::isFulfilled(const IFixedDecimal &number) {
     return result;
 }
 
-
 AndConstraint*
-AndConstraint::add()
-{
+AndConstraint::add(UErrorCode& status) {
+    if (U_FAILURE(fInternalStatus)) {
+        status = fInternalStatus;
+        return nullptr;
+    }
     this->next = new AndConstraint();
+    if (this->next == nullptr) {
+        status = U_MEMORY_ALLOCATION_ERROR;
+    }
     return this->next;
 }
 
+
 OrConstraint::OrConstraint() {
-    childNode=NULL;
-    next=NULL;
+    childNode = nullptr;
+    next = nullptr;
+    fInternalStatus = U_ZERO_ERROR;
 }
 
 OrConstraint::OrConstraint(const OrConstraint& other) {
-    if ( other.childNode == NULL ) {
-        this->childNode = NULL;
+    this->fInternalStatus = other.fInternalStatus;
+    if (U_FAILURE(fInternalStatus)) {
+        return; // stop early if the object we are copying from is invalid.
```

Comment:
Same as above, init pointer members before returning.

Reply:
Thank you again.